compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -3567,13 +3567,22 @@ void OMR::ValuePropagation::transformArrayCloneCall(TR::TreeTop *callTree, OMR::
    if (method->getRecognizedMethod() == TR::java_lang_J9VMInternals_primitiveClone)
       objNode = callNode->getLastChild();
 
-   if (!performTransformation(comp(), "%sInlining array clone call [%p] as new array and arraycopy\n", OPT_DETAILS, callNode))
-      return;
-
    TR_OpaqueClassBlock *j9arrayClass = cloneInfo->_clazz;
-   bool isFixedClass = cloneInfo->_isFixed;
-
    TR_OpaqueClassBlock *j9class = comp()->fe()->getComponentClassFromArrayClass(j9arrayClass);
+   bool isPrimitiveClass = TR::Compiler->cls.isPrimitiveClass(comp(), j9class);
+
+   if (isPrimitiveClass ? !cg()->getSupportsPrimitiveArrayCopy()
```

Comment:
sorry I missed this the first time, but conditional evaluation in an if condition is not ideal in this form. I would prefer:
```
if ((isPrimitiveClass && !cg()->getSupportsPrimitiveArrayCopy())
    || (!isPrimitiveClass && !cg()->getSupportsReferenceArrayCopy()))
```
as this would make the relationships more clear IMO

Reply:
Updated the code reflecting the review comment.